api/src/main/java/com/epam/pipeline/manager/preprocessing/NgsPreprocessingManager.java

```diff
@@ -0,0 +1,347 @@
+                        MessageConstants.ERROR_NGS_PREPROCESSING_FOLDER_SHOULD_HAVE_METADATA,
+                        projectIndicator
+                ));
+
+        final Pair<String, String> projectTypeIndicator =
+                fetchProjectIndicatorPair(SystemPreferences.UI_NGS_PROJECT_INDICATOR.getKey());
+        PipeConfValue projectTypeAttribute = folderMetadata.getData().get(projectTypeIndicator.getFirst());
+        Assert.state(projectTypeAttribute != null
+                        && projectTypeAttribute.getValue().equals(projectTypeIndicator.getSecond()),
+                messageHelper.getMessage(
+                        MessageConstants.ERROR_NGS_PREPROCESSING_FOLDER_SHOULD_HAVE_METADATA,
+                        projectTypeIndicator
+                ));
+
+        return folderMetadata;
+    }
+
+    private List<MetadataEntityVO> mapSampleSheetToMetadataEntities(final Long folderId, final SampleSheet sampleSheet,
+                                                                    final MetadataClass metadataClass,
+                                                                    final List<String> dataHeader) {
+        final int sampleIdIndex = dataHeader.indexOf(SampleSheetParser.SAMPLE_ID_COLUMN);
+        Assert.state(sampleIdIndex != -1,
+                messageHelper.getMessage(MessageConstants.ERROR_NGS_PREPROCESSING_SAMPLE_ID_NOT_FOUND));
+        final int laneIndex = dataHeader.indexOf(SampleSheetParser.LANE_COLUMN);
+
+        List<MetadataEntityVO> result = new ArrayList<>();
+        for (int i = 0; i < sampleSheet.getDataLines().size(); i++) {
+            String l = sampleSheet.getDataLines().get(i);
+            final List<String> fields = Arrays.asList(l.split(SampleSheetParser.SAMPLESHEET_DELIMETR));
+
+            final MetadataEntityVO entityVO = new MetadataEntityVO();
+
+            entityVO.setClassName(metadataClass.getName());
+            entityVO.setClassId(metadataClass.getId());
+            entityVO.setParentId(folderId);
+            if (laneIndex < 0) {
+                entityVO.setExternalId(fields.get(sampleIdIndex) + SAMPLE_PREFIX + i);
+            } else {
+                entityVO.setExternalId(fields.get(sampleIdIndex) + SAMPLE_PREFIX + i
+                        + LANE_PREFIX + fields.get(laneIndex));
+            }
+            final Map<String, PipeConfValue> data = new HashMap<>();
+            for (int j = 0; j < dataHeader.size(); j++) {
+                data.put(dataHeader.get(j),
+                        new PipeConfValue(
+                                PipeConfValueType.STRING.toString(),
+                                fields.size() > j ? fields.get(j) : StringUtils.EMPTY
+                        )
+                );
+            }
+            entityVO.setData(data);
+            result.add(entityVO);
+        }
+
+        return result;
+    }
+
+    private Pair<String, String> fetchProjectIndicatorPair(final String preferenceKey) {
```

Comment:
I think we have `parseProjectIndicator` similar method in `FolderManager`, let's extract it to some Utils method